File: frontend/src/app/internacao/models/cid.ts
export interface ICID {
    id?: number;
    codigo?: string;
    descricao?: string;
    pai?: ICID;
}

export class CID implements ICID {
    public id: number;
    public codigo: string;
    public descricao: string;
    public pai: CID;
}

File: prescricao/src/main/java/br/com/basis/madre/prescricao/domain/CID.java
package br.com.basis.madre.prescricao.domain;

import org.springframework.data.elasticsearch.annotations.Document;
import org.springframework.data.elasticsearch.annotations.Field;
import org.springframework.data.elasticsearch.annotations.FieldType;

import lombok.Data;
import lombok.NoArgsConstructor;

@Document(indexName = "madre-prescricao-cid")
@Data
@NoArgsConstructor
public class CID {

	@Field(type = FieldType.Long)
	private Long id;

	@Field(type = FieldType.Text)
	private String codigo;

	@Field(type = FieldType.Text)
	private String descricao;

}
